components/eamxx/src/share/field/field_utils.hpp

```diff
@@ -111,6 +112,12 @@ void perturb (const Field& f,
   impl::perturb<ST>(f, engine, pdf, base_seed, level_mask, dof_gids);
 }
 
+template <typename ST>
+Field dot_along_rank1_dim(const int &pd, const Field &f1, const Field &f2,
+                           const ekat::Comm *co = nullptr) {
+  return do_dot_along_rank1_dim<ST>(pd, f1, f2, co);
```

Comment:
I would say add to the implementation to the `impl` namespace like is done with the other utils.

Reply:
I also vote for using the impl namespace and keep the same signature.